File: frontend/src/types/email.ts
export interface EmailSummary {
  id: string;
  sender: string;
  subject: string;
  ingested_at: string;
  status: 'pending' | 'processing' | 'analyzed' | 'error';
  risk_score?: number;
}

export interface EmailUploadResponse {
  case_id?: string | null;
  email_id: string;
  status: string;
  hashes: { sha256: string; sha1?: string; md5?: string };
  ingested_at: string;
}

export interface EmailDetail extends EmailSummary {
  recipients: string[];
  body_text: string;
  body_html: string;
  headers: Record<string, string>;
  attachments: AttachmentInfo[];
  urls: string[];
}

export interface AttachmentInfo {
  filename: string;
  content_type: string;
  size: number;
  sha256: string;
}

File: backend/app/schemas/email.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional, Dict, Any
from datetime import datetime
from uuid import UUID
from app.models.email_case import EmailStatus

class EmailUploadResponse(BaseModel):
    case_id: Optional[UUID] = None
    email_id: UUID
    status: EmailStatus
    hashes: Dict[str, str]
    ingested_at: datetime
    model_config = ConfigDict(from_attributes=True)

class EmailSummary(BaseModel):
    id: UUID
    sender: Optional[str]
    subject: Optional[str]
    ingested_at: datetime
    status: EmailStatus
    risk_score: Optional[float] = None
    model_config = ConfigDict(from_attributes=True)

class EmailDetail(BaseModel):
    id: UUID
    sender: Optional[str]
    recipients: Optional[Any]
    subject: Optional[str]
    body_text: Optional[str]
    body_html: Optional[str]
    headers: Optional[Dict[str, Any]]
    attachments: Optional[Any]
    urls: Optional[Any]
    ingested_at: datetime
    status: EmailStatus
    model_config = ConfigDict(from_attributes=True)

class EmailListResponse(BaseModel):
    items: List[EmailSummary]
    total: int
    page: int
    page_size: int
